pkg/subctl/cmd/join.go

```diff
@@ -185,12 +189,45 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 	clusterCIDR, err = getPodCIDR(clusterCIDR, networkDetails)
 	exitOnError("Error determining the pod CIDR", err)
 
+	status.Start("* Discovering multi cluster details")
+	globalNetworks := getGlobalNetworks(subctlData)
+	err = checkOverlappingServiceCidr(globalNetworks)
+	status.End(err == nil)
+	exitOnError("Error validating overlapping CLusterCIDRs", err)
+
 	status.Start("Deploying Submariner")
 	err = submarinercr.Ensure(config, OperatorNamespace, populateSubmarinerSpec(subctlData))
 	status.End(err == nil)
 	exitOnError("Error deploying Submariner", err)
 }
 
+func checkOverlappingServiceCidr(networks map[string]*globalnet.GlobalNetwork) error {
+	for k, v := range networks {
+		overlap, err := globalnet.IsOverlappingCIDR(v.ServiceCIDRs, serviceCIDR)
```

Comment:
Why are we checking ONLY serviceCIDR? Ideally AFAIU (when not using globalnet) we should be checking both Service CIDR and Cluster CIDR. Otherwise, there could be issues when SM route-agent program rules where traffic destined to Overlapping Cluster/POD CIDR could be redirected over the tunnel. Even though LH supports the discovery of Services, we should check both POD/Service CIDRs.

Reply:
Wasn't sure if we want to check ClusterCIDRs or not. Will add.

Reply:
It'd be cool also to find a way to check for also the worker node IPs, may be we can add annotations about that info into the Cluster objects at some point, but we don't have it now.